test/test.pluggableauthhandler.ts

```diff
@@ -0,0 +1,455 @@
+        spawnStub,
+        expectedCommand,
+        expectedArgs,
+        expectedOpts
+      );
+    });
+
+    it('should call executable with correct command with spaces', async () => {
+      const handlerOptions = {
+        command: '"./command with/spaces.exe" -opt arg',
+        outputFile: 'output',
+        timeoutMillis: 1000,
+      } as PluggableAuthHandlerOptions;
+      const handler = new PluggableAuthHandler(handlerOptions);
+      const expectedCommand = './command with/spaces.exe';
+      const expectedArgs = ['-opt', 'arg'];
+
+      const response = handler.retrieveResponseFromExecutable(expectedEnvMap);
+      spawnEvent.stdout!.emit('data', JSON.stringify(defaultResponseJson));
+      spawnEvent.emit('close', 0);
+      await response;
+
+      sandbox.assert.calledOnceWithMatch(
+        spawnStub,
+        expectedCommand,
+        expectedArgs,
+        expectedOpts
+      );
+    });
+
+    it('should call executable with correct arguments with spaces', async () => {
+      const handlerOptions = {
+        command: './command/with/path.exe -opt "arg with spaces"',
+        outputFile: 'output',
+        timeoutMillis: 1000,
+      } as PluggableAuthHandlerOptions;
+      const handler = new PluggableAuthHandler(handlerOptions);
+      const expectedCommand = './command/with/path.exe';
+      const expectedArgs = ['-opt', 'arg with spaces'];
+
+      const response = handler.retrieveResponseFromExecutable(expectedEnvMap);
+      spawnEvent.stdout!.emit('data', JSON.stringify(defaultResponseJson));
+      spawnEvent.emit('close', 0);
+      await response;
+
+      sandbox.assert.calledOnceWithMatch(
+        spawnStub,
+        expectedCommand,
+        expectedArgs,
+        expectedOpts
+      );
+    });
+
+    it('should throw ExecutableError when executable fails', async () => {
+      const expectedError = new ExecutableError('test error', '1');
+      const handler = new PluggableAuthHandler(defaultHandlerOptions);
+
+      const response = handler.retrieveResponseFromExecutable(
+        new Map<string, string>()
+      );
+      spawnEvent.stderr!.emit('error', 'test error');
+      spawnEvent.emit('close', 1);
+
+      await assert.rejects(response, expectedError);
+    });
+
+    it('should throw error when executable times out', async () => {
+      const expectedError = new Error(
+        'The executable failed to finish within the timeout specified.'
+      );
+      spawnEvent.kill = () => {
+        return true;
+      };
+      const handler = new PluggableAuthHandler(defaultHandlerOptions);
+
+      const response = handler.retrieveResponseFromExecutable(
+        new Map<string, string>()
+      );
+      clock.tick(10001);
+
+      await assert.rejects(response, expectedError);
+    });
+
+    it('should throw error when non-json text is returned', async () => {
+      const expectedError = new Error(
+        'The executable returned an invalid response: Invalid response'
```

Comment:
Maybe "INVALID EXECUTABLE RESPONSE" + a comment that the executable response is appended? Took me a sec to realize you were testing that